hasher-matcher-actioner/tests/scripts/check_deployed_instance.py

```diff
@@ -0,0 +1,184 @@
+
+TEST_TE_DATA_PDQ = f"""0000000000000000000000000000000000000000000000000000000000000000,0000000000000001,2020-07-31T18:47:52+0000,tag1 tag2 tag3
+000000000000000000000000000000000000000000000000000000000000ffff,0000000000000001,2020-07-31T18:47:52+0000,tag1 tag2 tag3
+0f0f0f0f0f0f0f0f0f0f0f0f0f0f0f0f0f0f0f0f0f0f0f0f0f0f0f0f0f0f0f0f,0000000000000002,2020-07-31T18:47:52+0000,tag1 tag2 tag3
+f0f0f0f0f0f0f0f0f0f0f0f0f0f0f0f0f0f0f0f0f0f0f0f0f0f0f0f0f0f0f0f0,0000000000000003,2020-07-31T18:47:52+0000,tag1 tag2 tag3
+ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff,0000000000000004,2020-07-31T18:47:52+0000,tag1 tag2 tag3
+1111111111111111111111111111111111111111111111111111111111111111,0000000000000005,2020-07-31T18:47:52+0000,tag1 tag2 tag3
+{TEST_PHOTO_EXPECTED_HASH},{TEST_PHOTO_EXPECTED_ID},2020-07-31T18:47:52+0000,tag1 tag2 tag3
+"""
+
+
+class TestError(Exception):
+    """
+    Wrapper for exceptions which cause return codes
+    """
+    def __init__(self, message: str, returncode: int = 1) -> None:
+        super().__init__(message)
+        self.returncode = returncode
+
+
+def upload_s3_object(bucket: str, key: str, body: str) -> None:
+    """
+    Uploads an object,
+    throws if upload fails (put_object only returns if succesful).
+    """
+    try:
+        print(f"Attempting to upload {key}...")
+        s3_client.put_object(Bucket=bucket, Key=key, Body=body)
+    except ClientError as e:
+        raise TestError(e.msg)
+
+
+def wait_for_s3_object(bucket: str, key: str, wait_time=5, retries=20) -> None:
+    """
+    Waits for specfic object in given s3 bucket.
+    """
+    try:
+        print(f"Looking for {key} in {bucket} (check every {wait_time}s x{retries})...")
+        waiter = s3_client.get_waiter("object_exists")
+        waiter.wait(
+            Bucket=bucket,
+            Key=key,
+            WaiterConfig={"Delay": wait_time, "MaxAttempts": retries},
+        )
+    except ClientError as e:
+        raise TestError(e.msg)
+
+
+def wait_for_db_item(
+    table, key: t.Dict, attributes: t.List, wait_time=5, retries=20
+) -> t.Dict:
+    """
+    Query in a retry loop for a specfic item in table.
+    """
+    try:
+        print(f"Looking for {key} in {table} (check every {wait_time}s x{retries})...")
+        while retries > 0:
+            result = table.get_item(
+                Key=key,
+                AttributesToGet=attributes,
+            )
+            if "Item" in result:
+                return result["Item"]
+            time.sleep(wait_time)
+            retries -= 1
+        return {}
+    except ClientError as e:
+        raise TestError(e.msg)
+
+
+def run(bucket: str, table) -> bool:
+    """
+    ~End-to-end test of HMA that:
+    uploads TE data -> looks for the index
+    uploads a photo -> looks for the hash and a match
+
+    throws TestError if it encoutners unexpected results
+    """
+    # Upload TE Data
+    upload_s3_object(bucket, THREAT_EXCHANGE_PDQ_DATA_KEY, TEST_TE_DATA_PDQ)
```

Comment:
If I am testing an instance the second time, this object will already exist. In those cases, can we check that the index exists and proceed? 

Or perhaps you think that is not an end-to-end test. What do you think?